src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -701,19 +847,26 @@ internal sealed class TupleFieldSymbol : FieldSymbol
         /// A tuple without backing fields is usable for binding purposes, since we know its name and type,
         /// but caller is supposed to report some kind of error at declaration.
         /// </summary>
-        public TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, FieldSymbol underlyingFieldOpt)
+        public TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, OneOrMany<FieldSymbol> underlyingFieldChainOpt)
```

Comment:
We should not optimize for gigantic tuples (especially if at the cost to small ones).
Lets try just keep the top level type and do everything based off the field's ordinal position.